File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    // count := 0
    var element, count int;

    for _, v := range nums{
        if count == 0{
            element = v
        }
        
        if v == element {
            count++
        } else{
            count--
        }
    }

    return element
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int element, count = 0;
        for(int i=0; i<nums.size(); i++){
            if(count==0){
                count = 1;
                element = nums[i];
            } 
            else if(nums[i] == element){
                count++;
            }
            else{
                count--;
            }
        }

        return element;
    }
};